app/lib/search/backend.dart

```diff
@@ -138,6 +147,71 @@ class SearchBackend {
   }
 }
 
+class PopularityStorage {
+  final Storage storage;
+  final Bucket bucket;
+
+  final Map<String, double> _values = {};
+
+  PopularityStorage(this.storage, this.bucket);
+
+  double operator [](String package) => _values[package];
+
+  Future init() async {
+    await fetch();
+    new Timer.periodic(const Duration(days: 1), (_) {
+      fetch();
+    });
+  }
+
+  Future fetch() async {
+    _logger.info('Updating popularity storage...');
+    final List<BucketEntry> list = await bucket.list().toList();
```

Comment:
Please wrap this in try-catch and re-try. If the first call has an error, e.g. a timeout, then the VM will never get the scores.

I cannot recall if `package:gcloud` has specific exception classes, so we can avoid re-trying if e.g. the bucket doesn't exist or we don't have permission. If so, we could special case on the exception type.